src/app/clusters/energy-preference-server/energy-preference-server.h

```diff
@@ -0,0 +1,44 @@
+/**
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <stddef.h>
+
+#include <app-common/zap-generated/cluster-enums.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/util/basic-types.h>
+#include <lib/core/CHIPError.h>
+
+struct EnergyPreferenceDelegate
```

Comment:
ok, I've just been using Identify as the model for how to implement these things (and yeah, didn't add any doxygen stuff), is there a implementation example you would prefer I use for namespacing?  Otherwise I'll just stick it in the same name space as the server impl.